### ISSUE TITLE
GRAPPA result is uint16

### ISSUE THREAD

**Issue by zaccharieramzi** [gadgetron/gadgetron#872]
Hi,

I am trying to perform GRAPPA reconstruction using gadgetron (with docker).
I want to do this on retrospectively undersampled data, so I have to create the ismrmrd file manually using [`ismrmrd-python`](https://github.com/ismrmrd/ismrmrd-python).
The dataset I am using is [fastmri](https://fastmri.org/).

I managed to create [this file](https://github.com/gadgetron/gadgetron/files/4925301/file.zip), but I am not sure that it fits all the correct formatting.

When using the default reconstruction, with `gadgetron_ismrmrd_client -f file1000000_slice_14.h5 -o out_slice_14_default.h5 -c default.xml -v -G recon_default`, I have a correct reconstruction, which looks like this (the acceleration factor is 4 so this is expected):
![default_gadgetron_recon](https://user-images.githubusercontent.com/6387497/87548145-5f135b00-c6ac-11ea-8243-e0b85dbe94ae.png)

However, when using the GRAPPA reconstruction with `gadgetron_ismrmrd_client -f file1000000_slice_14.h5 -o out_slice_14.h5 -c Generic_Cartesian_Grappa.xml -v -G recon_grappa`, the reconstruction I get is entirely black and of dtype `uint16` (not showing the image because it's all black anyway filled with 0s).

The output on the client side is:
```
Gadgetron ISMRMRD client
  -- host            :      localhost
  -- port            :      9002
  -- hdf5 file  in   :      file1000000_slice_14.h5
  -- hdf5 group in   :      /dataset
  -- conf            :      Generic_Cartesian_Grappa.xml
  -- loop            :      1
  -- hdf5 file out   :      out_slice_14.h5
  -- hdf5 group out  :      recon_grappa
Find 97 ismrmrd acquisitions
Find 0 ismrmrd waveforms
```

The output in the logs for the server is:
```
07-15 12:31:21.448 INFO [GadgetStreamController.cpp:48] Connection from 127.0.0.1:40238
07-15 12:31:21.448 INFO [GadgetStreamController.cpp:202] Running configuration: /usr/local/share/gadgetron/config/Generic_Cartesian_Grappa.xml
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:226] Found 2 readers
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:227] Found 1 writers
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:228] Found 16 gadgets
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:244] --Found reader declaration
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:245]   Reader dll: gadgetron_mricore
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:246]   Reader class: GadgetIsmrmrdAcquisitionMessageReader
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:247]   Reader slot: 1008
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:244] --Found reader declaration
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:245]   Reader dll: gadgetron_mricore
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:246]   Reader class: GadgetIsmrmrdWaveformMessageReader
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:247]   Reader slot: 1026
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:277] --Found writer declaration
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:278]   Writer dll: gadgetron_mricore
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:279]   Writer class: MRIImageWriter
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:280]   Writer slot: 1022
07-15 12:31:21.449 DEBUG [GadgetStreamController.cpp:296] Processing 16 gadgets in reverse order
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:311]   Gadget Name: ImageFinish
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:313]   Gadget class: ImageFinishGadget
07-15 12:31:21.449 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:311]   Gadget Name: FloatToShortAttrib
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:313]   Gadget class: FloatToUShortGadget
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 3
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:335] Setting parameter max_intensity = 32767
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:335] Setting parameter min_intensity = 0
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:335] Setting parameter intensity_offset = 0
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:311]   Gadget Name: ComplexToFloatAttrib
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:313]   Gadget class: ComplexToFloatGadget
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:311]   Gadget Name: ImageArraySplit
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:313]   Gadget class: ImageArraySplitGadget
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 12:31:21.450 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:311]   Gadget Name: Scaling
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:313]   Gadget class: GenericReconImageArrayScalingGadget
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 8
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:335] Setting parameter perform_timing = false
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:335] Setting parameter verbose = false
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:335] Setting parameter min_intensity_value = 64
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:335] Setting parameter max_intensity_value = 4095
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:335] Setting parameter scalingFactor = 10.0
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:335] Setting parameter use_constant_scalingFactor = true
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:335] Setting parameter auto_scaling_only_once = true
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:335] Setting parameter scalingFactor_dedicated = 100.0
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:311]   Gadget Name: FOVAdjustment
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:313]   Gadget class: GenericReconFieldOfViewAdjustmentGadget
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 3
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:335] Setting parameter debug_folder = 
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:335] Setting parameter perform_timing = false
07-15 12:31:21.451 INFO [GadgetStreamController.cpp:335] Setting parameter verbose = false
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:311]   Gadget Name: KSpaceFilter
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:313]   Gadget class: GenericReconKSpaceFilteringGadget
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 13
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter debug_folder = 
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter perform_timing = false
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter verbose = false
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter skip_processing_meta_field = Skip_processing_after_recon
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter filterRO = Gaussian
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter filterRO_sigma = 1.0
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter filterRO_width = 0.15
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter filterE1 = Gaussian
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter filterE1_sigma = 1.0
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter filterE1_width = 0.15
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter filterE2 = Gaussian
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter filterE2_sigma = 1.0
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:335] Setting parameter filterE2_width = 0.15
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:311]   Gadget Name: PartialFourierHandling
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.452 INFO [GadgetStreamController.cpp:313]   Gadget class: GenericReconPartialFourierHandlingFilterGadget
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 8
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter debug_folder = 
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter perform_timing = false
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter verbose = false
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter skip_processing_meta_field = Skip_processing_after_recon
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter partial_fourier_filter_RO_width = 0.15
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter partial_fourier_filter_E1_width = 0.15
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter partial_fourier_filter_E2_width = 0.15
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter partial_fourier_filter_densityComp = false
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:311]   Gadget Name: Recon
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:313]   Gadget class: GenericReconCartesianGrappaGadget
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 9
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter image_series = 0
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter coil_map_algorithm = Inati
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter downstream_coil_compression = true
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter downstream_coil_compression_thres = 0.01
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter downstream_coil_compression_num_modesKept = 0
07-15 12:31:21.453 INFO [GadgetStreamController.cpp:335] Setting parameter debug_folder = 
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:335] Setting parameter perform_timing = true
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:335] Setting parameter verbose = true
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:335] Setting parameter send_out_gfactor = false
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:311]   Gadget Name: CoilCompression
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:313]   Gadget class: GenericReconEigenChannelGadget
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 8
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:335] Setting parameter debug_folder = 
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:335] Setting parameter perform_timing = true
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:335] Setting parameter verbose = true
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:335] Setting parameter average_all_ref_N = true
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:335] Setting parameter average_all_ref_S = true
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:335] Setting parameter upstream_coil_compression = true
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:335] Setting parameter upstream_coil_compression_thres = 0.002
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:335] Setting parameter upstream_coil_compression_num_modesKept = 0
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:311]   Gadget Name: PrepRef
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:313]   Gadget class: GenericReconCartesianReferencePrepGadget
07-15 12:31:21.454 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 6
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:335] Setting parameter debug_folder = 
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:335] Setting parameter perform_timing = true
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:335] Setting parameter verbose = true
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:335] Setting parameter average_all_ref_N = true
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:335] Setting parameter average_all_ref_S = true
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:335] Setting parameter prepare_ref_always = true
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:311]   Gadget Name: BucketToBuffer
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:313]   Gadget class: BucketToBufferGadget
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 5
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:335] Setting parameter N_dimension = contrast
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:335] Setting parameter S_dimension = average
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:335] Setting parameter split_slices = false
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:335] Setting parameter ignore_segment = true
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:335] Setting parameter verbose = true
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:311]   Gadget Name: AccTrig
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:313]   Gadget class: AcquisitionAccumulateTriggerGadget
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 2
07-15 12:31:21.455 INFO [GadgetStreamController.cpp:335] Setting parameter trigger_dimension = 
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:335] Setting parameter sorting_dimension = 
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:311]   Gadget Name: RemoveROOversampling
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:313]   Gadget class: RemoveROOversamplingGadget
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:311]   Gadget Name: AsymmetricEcho
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:313]   Gadget class: AsymmetricEchoAdjustROGadget
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:311]   Gadget Name: NoiseAdjust
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:313]   Gadget class: NoiseAdjustGadget
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 12:31:21.456 INFO [GadgetStreamController.cpp:356] Gadget Stream configured
07-15 12:31:21.457 DEBUG [NoiseAdjustGadget.cpp:58] Folder to store noise dependencies is /tmp/gadgetron/
07-15 12:31:21.457 DEBUG [NoiseAdjustGadget.cpp:63] NoiseAdjustGadget::perform_noise_adjust_ is 1
07-15 12:31:21.457 DEBUG [NoiseAdjustGadget.cpp:66] NoiseAdjustGadget::pass_nonconformant_data_ is 0
07-15 12:31:21.457 DEBUG [NoiseAdjustGadget.cpp:75] receiver_noise_bandwidth_ is 0.793000
07-15 12:31:21.457 DEBUG [NoiseAdjustGadget.cpp:84] Measurement ID is 67041_36245722_36245731_3604
07-15 12:31:21.457 DEBUG [AsymmetricEchoAdjustROGadget.cpp:23] max RO for encoding space  0 : 640
07-15 12:31:21.458 DEBUG [RemoveROOversamplingGadget.cpp:43] RemoveROOversamplingGadget:omp_set_num_threads(1) ... 
07-15 12:31:21.458 DEBUG [AcquisitionAccumulateTriggerGadget.cpp:92] TRIGGER DIMENSION IS:  (18)
07-15 12:31:21.458 DEBUG [AcquisitionAccumulateTriggerGadget.cpp:156] SORTING DIMENSION IS:  (18)
07-15 12:31:21.458 DEBUG [BucketToBufferGadget.cpp:49] N DIMENSION IS: contrast (4)
07-15 12:31:21.458 DEBUG [BucketToBufferGadget.cpp:80] S DIMENSION IS: average (2)
07-15 12:31:21.458 DEBUG [BucketToBufferGadget.cpp:83] SPLIT SLICES IS: 0
07-15 12:31:21.458 DEBUG [BucketToBufferGadget.cpp:86] IGNORE SEGMENT IS: 1
07-15 12:31:21.459 DEBUG [GenericReconBase.cpp:41] Debug folder is not set ... 
07-15 12:31:21.459 DEBUG [GenericReconCartesianReferencePrepGadget.cpp:42] Number of encoding spaces: 1
07-15 12:31:21.459 DEBUG [GenericReconCartesianReferencePrepGadget.cpp:53] ---> Encoding space : 0 <---
07-15 12:31:21.459 DEBUG [GenericReconCartesianReferencePrepGadget.cpp:54] Encoding matrix size: 640 368 1
07-15 12:31:21.459 DEBUG [GenericReconCartesianReferencePrepGadget.cpp:55] Encoding field_of_view : 280 161.42 4.5
07-15 12:31:21.459 DEBUG [GenericReconCartesianReferencePrepGadget.cpp:56] Recon matrix size : 320 320 1
07-15 12:31:21.459 DEBUG [GenericReconCartesianReferencePrepGadget.cpp:57] Recon field_of_view :  140 140 3
07-15 12:31:21.459 DEBUG [GenericReconCartesianReferencePrepGadget.cpp:68] acceFactorE1 is 1
07-15 12:31:21.459 DEBUG [GenericReconCartesianReferencePrepGadget.cpp:69] acceFactorE2 is 1
07-15 12:31:21.459 DEBUG [GenericReconBase.cpp:41] Debug folder is not set ... 
07-15 12:31:21.460 DEBUG [GenericReconEigenChannelGadget.cpp:42] Number of encoding spaces: 1
07-15 12:31:21.460 DEBUG [GenericReconBase.cpp:41] Debug folder is not set ... 
07-15 12:31:21.460 DEBUG [ImageArraySendMixin.hpp:158] ---> Encoding space : 0 <---
07-15 12:31:21.460 DEBUG [ImageArraySendMixin.hpp:161] Encoding matrix size: 640 368 1
07-15 12:31:21.460 DEBUG [ImageArraySendMixin.hpp:164] Encoding field_of_view : 280 161.42 4.5
07-15 12:31:21.460 DEBUG [ImageArraySendMixin.hpp:167] Recon matrix size : 320 320 1
07-15 12:31:21.460 DEBUG [ImageArraySendMixin.hpp:170] Recon field_of_view :  140 140 3
07-15 12:31:21.460 DEBUG [GenericReconGadget.cpp:41] Number of encoding spaces: 1
07-15 12:31:21.460 DEBUG [GenericReconGadget.cpp:68] acceFactorE1 is 1
07-15 12:31:21.460 DEBUG [GenericReconGadget.cpp:69] acceFactorE2 is 1
07-15 12:31:21.460 DEBUG [GenericReconCartesianGrappaGadget.cpp:39] Number of encoding spaces: 1
07-15 12:31:21.486 DEBUG [Gadget.h:174] Gadget (NoiseAdjust) Close Called with flags = 1
07-15 12:31:21.486 DEBUG [Gadget.h:184] Gadget (NoiseAdjust) waiting for thread to finish
07-15 12:31:21.486 DEBUG [Gadget.h:174] Gadget (NoiseAdjust) Close Called with flags = 0
07-15 12:31:21.487 DEBUG [Gadget.h:186] Gadget (NoiseAdjust) thread finished
07-15 12:31:21.487 DEBUG [Gadget.h:123] Shutting down Gadget (NoiseAdjust)
07-15 12:31:21.487 DEBUG [Gadget.h:174] Gadget (AsymmetricEcho) Close Called with flags = 1
07-15 12:31:21.487 DEBUG [Gadget.h:184] Gadget (AsymmetricEcho) waiting for thread to finish
07-15 12:31:21.487 DEBUG [Gadget.h:174] Gadget (AsymmetricEcho) Close Called with flags = 0
07-15 12:31:21.487 DEBUG [Gadget.h:186] Gadget (AsymmetricEcho) thread finished
07-15 12:31:21.487 DEBUG [Gadget.h:123] Shutting down Gadget (AsymmetricEcho)
07-15 12:31:21.487 DEBUG [Gadget.h:174] Gadget (RemoveROOversampling) Close Called with flags = 1
07-15 12:31:21.487 DEBUG [Gadget.h:184] Gadget (RemoveROOversampling) waiting for thread to finish
07-15 12:31:21.487 DEBUG [Gadget.h:174] Gadget (RemoveROOversampling) Close Called with flags = 0
07-15 12:31:21.487 DEBUG [Gadget.h:186] Gadget (RemoveROOversampling) thread finished
07-15 12:31:21.487 DEBUG [Gadget.h:123] Shutting down Gadget (RemoveROOversampling)
07-15 12:31:21.487 DEBUG [Gadget.h:174] Gadget (AccTrig) Close Called with flags = 1
07-15 12:31:21.487 DEBUG [Gadget.h:184] Gadget (AccTrig) waiting for thread to finish
07-15 12:31:21.487 DEBUG [Gadget.h:174] Gadget (AccTrig) Close Called with flags = 0
07-15 12:31:21.487 DEBUG [Gadget.h:186] Gadget (AccTrig) thread finished
07-15 12:31:21.487 DEBUG [AcquisitionAccumulateTriggerGadget.cpp:495] AcquisitionAccumulateTriggerGadget::close
07-15 12:31:21.487 DEBUG [AcquisitionAccumulateTriggerGadget.cpp:462] Trigger (1) occurred, sending out 1 buckets
07-15 12:31:21.488 DEBUG [Gadget.h:123] Shutting down Gadget (AccTrig)
07-15 12:31:21.488 DEBUG [Gadget.h:174] Gadget (BucketToBuffer) Close Called with flags = 1
07-15 12:31:21.488 DEBUG [BucketToBufferGadget.cpp:735] Encoding space : 0 - FOV : [ 280 161.42 4.5 ]  - Matris size : [ 640 368 1 ] 
07-15 12:31:21.488 DEBUG [Gadget.h:184] Gadget (BucketToBuffer) waiting for thread to finish
07-15 12:31:21.488 DEBUG [BucketToBufferGadget.cpp:740] Sampling limits : - RO : [ 0 160 319 ] - E1 : [ 0 184 368 ] - E2 : [ 0 0 0 ]
07-15 12:31:21.488 DEBUG [BucketToBufferGadget.cpp:618] Data dimensions [RO E1 E2 CHA N S SLC] : [320 368 1 15 1 1 1]
07-15 12:31:21.509 DEBUG [BucketToBufferGadget.cpp:222] End of bucket reached, sending out 1 ReconData buffers
07-15 12:31:21.509 DEBUG [Gadget.h:174] Gadget (BucketToBuffer) Close Called with flags = 0
07-15 12:31:21.509 DEBUG [BucketToBufferGadget.cpp:276] BucketToBufferGadget::close
07-15 12:31:21.509 DEBUG [Gadget.h:186] Gadget (BucketToBuffer) thread finished
07-15 12:31:21.509 DEBUG [BucketToBufferGadget.cpp:276] BucketToBufferGadget::close
07-15 12:31:21.509 DEBUG [Gadget.h:123] Shutting down Gadget (BucketToBuffer)
07-15 12:31:21.509 DEBUG [Gadget.h:174] Gadget (PrepRef) Close Called with flags = 1
07-15 12:31:21.510 DEBUG [Gadget.h:184] Gadget (PrepRef) waiting for thread to finish
07-15 12:31:21.567 DEBUG [GenericReconEigenChannelGadget.cpp:121] GenericReconEigenChannelGadget - incoming data array : [RO E1 E2 CHA N S SLC] - [320 368 1 15 1 1 1]
07-15 12:31:21.567 DEBUG [GadgetronTimer.h:87] GenericReconCartesianReferencePrepGadget::process:57.931000 ms
07-15 12:31:21.578 DEBUG [Gadget.h:174] Gadget (PrepRef) Close Called with flags = 0
07-15 12:31:21.578 DEBUG [Gadget.h:186] Gadget (PrepRef) thread finished
07-15 12:31:21.578 DEBUG [Gadget.h:123] Shutting down Gadget (PrepRef)
07-15 12:31:21.578 DEBUG [Gadget.h:174] Gadget (CoilCompression) Close Called with flags = 1
07-15 12:31:21.578 DEBUG [Gadget.h:184] Gadget (CoilCompression) waiting for thread to finish
07-15 12:31:21.709 DEBUG [GenericReconEigenChannelGadget.cpp:191] GenericReconEigenChannelGadget - Number of modes kept: 15 out of 15; Eigen value, slc - 0, S - 0, N - 0 : [
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (4.37527e-05,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (1.13985e-05,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (8.33646e-06,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (3.3338e-06,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (1.64567e-06,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (9.72226e-07,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (5.42225e-07,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (3.67581e-07,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (3.31552e-07,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (3.12581e-07,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (2.72251e-07,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (2.15171e-07,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (1.98195e-07,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (1.56112e-07,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:195]         (1.41223e-07,0)
07-15 12:31:21.710 DEBUG [GenericReconEigenChannelGadget.cpp:197] ]
07-15 12:31:21.825 DEBUG [GadgetronTimer.h:87] GenericReconEigenChannelGadget::process:258.206000 ms
07-15 12:31:21.825 DEBUG [Gadget.h:174] Gadget (CoilCompression) Close Called with flags = 0
07-15 12:31:21.825 DEBUG [GenericReconCartesianGrappaGadget.cpp:73] Calling 1 , encoding space : 0
07-15 12:31:21.826 DEBUG [GenericReconCartesianGrappaGadget.cpp:75] ======================================================================
07-15 12:31:21.826 DEBUG [Gadget.h:186] Gadget (CoilCompression) thread finished
07-15 12:31:21.826 DEBUG [Gadget.h:123] Shutting down Gadget (CoilCompression)
07-15 12:31:21.826 DEBUG [GenericReconCartesianGrappaGadget.cpp:719] GenericReconCartesianGrappaGadget - close(flags) : 1
07-15 12:31:21.826 DEBUG [Gadget.h:174] Gadget (Recon) Close Called with flags = 1
07-15 12:31:21.826 DEBUG [Gadget.h:184] Gadget (Recon) waiting for thread to finish
07-15 12:31:21.906 DEBUG [GadgetronTimer.h:87] GenericReconCartesianGrappaGadget::make_ref_coil_map:80.031000 ms
07-15 12:31:21.907 DEBUG [GenericReconCartesianGrappaGadget.cpp:346] Downstream coil compression is prescribed to use 7 out of 15 channels ...
07-15 12:31:21.911 DEBUG [GadgetronTimer.h:87] GenericReconCartesianGrappaGadget::prepare_down_stream_coil_compression_ref_data:5.295000 ms
07-15 12:31:22.006 DEBUG [GadgetronTimer.h:87] GenericReconCartesianGrappaGadget::perform_coil_map_estimation:94.273000 ms
07-15 12:31:22.021 DEBUG [GadgetronTimer.h:87] GenericReconCartesianGrappaGadget::perform_calib:15.194000 ms
07-15 12:31:22.083 DEBUG [GenericReconGadget.cpp:652] length for RO : 320 - 1
07-15 12:31:22.084 DEBUG [GenericReconGadget.cpp:655] effective_acce_factor : 4.04396
07-15 12:31:22.084 DEBUG [GenericReconGadget.cpp:661] snr_scaling_ratio : 2.01096
07-15 12:31:22.093 DEBUG [GenericReconCartesianGrappaGadget.cpp:615] GenericReconCartesianGrappaGadget, grappaKernelCompensationFactor*snr_scaling_ratio : 2.01096
07-15 12:31:22.102 DEBUG [GadgetronTimer.h:87] GenericReconCartesianGrappaGadget::perform_unwrapping:77.748000 ms
07-15 12:31:22.102 DEBUG [GadgetronTimer.h:87] GenericReconCartesianGrappaGadget::compute_image_header:0.199000 ms
07-15 12:31:22.102 DEBUG [ImageArraySendMixin.hpp:106] sending out image array, acquisition boundary [RO E1 E2 CHA N S SLC] = [320 368 1 1 1 1 1] 
07-15 12:31:22.102 DEBUG [ImageArraySendMixin.hpp:122] sending out Image image [CHA SLC CON PHS REP SET AVE] = [0 0 0 0 0 0 0 ]  -- Image series -- 1 -- Image number -- 1
07-15 12:31:22.102 DEBUG [GadgetronTimer.h:87] GenericReconCartesianGrappaGadget::send_out_image_array:0.474000 ms
07-15 12:31:22.103 DEBUG [GadgetronTimer.h:87] GenericReconCartesianGrappaGadget::process:277.115000 ms
07-15 12:31:22.103 DEBUG [GenericReconCartesianGrappaGadget.cpp:719] GenericReconCartesianGrappaGadget - close(flags) : 0
07-15 12:31:22.103 DEBUG [Gadget.h:174] Gadget (Recon) Close Called with flags = 0
07-15 12:31:22.103 DEBUG [Gadget.h:186] Gadget (Recon) thread finished
07-15 12:31:22.103 DEBUG [GenericReconCartesianGrappaGadget.cpp:726] Clean recon_obj_ for encoding space 0
07-15 12:31:22.104 DEBUG [Gadget.h:123] Shutting down Gadget (Recon)
07-15 12:31:22.104 DEBUG [GenericReconPartialFourierHandlingGadget.cpp:285] GenericReconPartialFourierHandlingGadget - close(flags) : 1
07-15 12:31:22.104 DEBUG [Gadget.h:174] Gadget (PartialFourierHandling) Close Called with flags = 1
07-15 12:31:22.104 DEBUG [Gadget.h:184] Gadget (PartialFourierHandling) waiting for thread to finish
07-15 12:31:22.105 DEBUG [GenericReconPartialFourierHandlingGadget.cpp:285] GenericReconPartialFourierHandlingGadget - close(flags) : 0
07-15 12:31:22.105 DEBUG [Gadget.h:174] Gadget (PartialFourierHandling) Close Called with flags = 0
07-15 12:31:22.105 DEBUG [Gadget.h:186] Gadget (PartialFourierHandling) thread finished
07-15 12:31:22.105 DEBUG [Gadget.h:123] Shutting down Gadget (PartialFourierHandling)
07-15 12:31:22.105 DEBUG [GenericReconKSpaceFilteringGadget.cpp:454] GenericReconKSpaceFilteringGadget - close(flags) : 1
07-15 12:31:22.105 DEBUG [Gadget.h:174] Gadget (KSpaceFilter) Close Called with flags = 1
07-15 12:31:22.105 DEBUG [Gadget.h:184] Gadget (KSpaceFilter) waiting for thread to finish
07-15 12:31:22.127 DEBUG [GenericReconKSpaceFilteringGadget.cpp:454] GenericReconKSpaceFilteringGadget - close(flags) : 0
07-15 12:31:22.127 DEBUG [Gadget.h:174] Gadget (KSpaceFilter) Close Called with flags = 0
07-15 12:31:22.127 DEBUG [Gadget.h:186] Gadget (KSpaceFilter) thread finished
07-15 12:31:22.127 DEBUG [Gadget.h:123] Shutting down Gadget (KSpaceFilter)
07-15 12:31:22.127 DEBUG [GenericReconFieldOfViewAdjustmentGadget.cpp:306] GenericReconFieldOfViewAdjustmentGadget - close(flags) : 1
07-15 12:31:22.127 DEBUG [Gadget.h:174] Gadget (FOVAdjustment) Close Called with flags = 1
07-15 12:31:22.127 DEBUG [Gadget.h:184] Gadget (FOVAdjustment) waiting for thread to finish
07-15 12:31:22.152 DEBUG [GenericReconFieldOfViewAdjustmentGadget.cpp:306] GenericReconFieldOfViewAdjustmentGadget - close(flags) : 0
07-15 12:31:22.152 DEBUG [Gadget.h:174] Gadget (FOVAdjustment) Close Called with flags = 0
07-15 12:31:22.156 DEBUG [Gadget.h:186] Gadget (FOVAdjustment) thread finished
07-15 12:31:22.156 DEBUG [Gadget.h:123] Shutting down Gadget (FOVAdjustment)
07-15 12:31:22.156 DEBUG [GenericReconImageArrayScalingGadget.cpp:235] GenericReconImageArrayScalingGadget - close(flags) : 1
07-15 12:31:22.156 DEBUG [Gadget.h:174] Gadget (Scaling) Close Called with flags = 1
07-15 12:31:22.156 DEBUG [Gadget.h:184] Gadget (Scaling) waiting for thread to finish
07-15 12:31:22.156 DEBUG [GenericReconImageArrayScalingGadget.cpp:235] GenericReconImageArrayScalingGadget - close(flags) : 0
07-15 12:31:22.156 DEBUG [Gadget.h:174] Gadget (Scaling) Close Called with flags = 0
07-15 12:31:22.156 DEBUG [Gadget.h:186] Gadget (Scaling) thread finished
07-15 12:31:22.156 DEBUG [Gadget.h:123] Shutting down Gadget (Scaling)
07-15 12:31:22.156 DEBUG [Gadget.h:174] Gadget (ImageArraySplit) Close Called with flags = 1
07-15 12:31:22.156 DEBUG [Gadget.h:184] Gadget (ImageArraySplit) waiting for thread to finish
07-15 12:31:22.156 DEBUG [Gadget.h:174] Gadget (ImageArraySplit) Close Called with flags = 0
07-15 12:31:22.156 DEBUG [Gadget.h:186] Gadget (ImageArraySplit) thread finished
07-15 12:31:22.156 DEBUG [Gadget.h:123] Shutting down Gadget (ImageArraySplit)
07-15 12:31:22.156 DEBUG [Gadget.h:174] Gadget (ComplexToFloatAttrib) Close Called with flags = 1
07-15 12:31:22.156 DEBUG [Gadget.h:184] Gadget (ComplexToFloatAttrib) waiting for thread to finish
07-15 12:31:22.175 DEBUG [Gadget.h:174] Gadget (ComplexToFloatAttrib) Close Called with flags = 0
07-15 12:31:22.176 DEBUG [Gadget.h:186] Gadget (ComplexToFloatAttrib) thread finished
07-15 12:31:22.176 DEBUG [Gadget.h:123] Shutting down Gadget (ComplexToFloatAttrib)
07-15 12:31:22.176 DEBUG [Gadget.h:174] Gadget (FloatToShortAttrib) Close Called with flags = 1
07-15 12:31:22.176 DEBUG [Gadget.h:184] Gadget (FloatToShortAttrib) waiting for thread to finish
07-15 12:31:22.188 DEBUG [Gadget.h:174] Gadget (FloatToShortAttrib) Close Called with flags = 0
07-15 12:31:22.189 DEBUG [Gadget.h:186] Gadget (FloatToShortAttrib) thread finished
07-15 12:31:22.189 DEBUG [Gadget.h:123] Shutting down Gadget (FloatToShortAttrib)
07-15 12:31:22.189 DEBUG [Gadget.h:174] Gadget (ImageFinish) Close Called with flags = 1
07-15 12:31:22.189 DEBUG [Gadget.h:184] Gadget (ImageFinish) waiting for thread to finish
07-15 12:31:22.189 DEBUG [Gadget.h:174] Gadget (ImageFinish) Close Called with flags = 0
07-15 12:31:22.190 DEBUG [Gadget.h:186] Gadget (ImageFinish) thread finished
07-15 12:31:22.190 DEBUG [Gadget.h:123] Shutting down Gadget (ImageFinish)
07-15 12:31:22.190 DEBUG [EndGadget.cpp:12] Close called in EndGadget with flags 1
07-15 12:31:22.190 DEBUG [EndGadget.cpp:23] Calling close in base class  with flags 1
07-15 12:31:22.190 DEBUG [Gadget.h:174] Gadget (EndGadget) Close Called with flags = 1
07-15 12:31:22.190 DEBUG [Gadget.h:184] Gadget (EndGadget) waiting for thread to finish
07-15 12:31:22.190 DEBUG [EndGadget.cpp:12] Close called in EndGadget with flags 0
07-15 12:31:22.190 DEBUG [EndGadget.cpp:23] Calling close in base class  with flags 0
07-15 12:31:22.190 DEBUG [Gadget.h:174] Gadget (EndGadget) Close Called with flags = 0
07-15 12:31:22.190 DEBUG [Gadget.h:186] Gadget (EndGadget) thread finished
07-15 12:31:22.190 DEBUG [Gadget.h:123] Shutting down Gadget (EndGadget)
07-15 12:31:22.190 DEBUG [GadgetStreamController.cpp:80] Stream closed
07-15 12:31:22.190 DEBUG [GadgetStreamController.cpp:81] Closing writer task
07-15 12:31:22.190 DEBUG [GadgetStreamController.cpp:83] Writer task closed
07-15 12:31:22.191 ERROR [GadgetStreamController.cpp:74] GadgetStreamController, unable to read message identifier
07-15 12:31:22.191 DEBUG [GadgetStreamController.cpp:159] handle_close called
07-15 12:31:22.191 INFO [GadgetStreamController.cpp:164] Shutting down stream and closing up shop...
07-15 12:31:22.191 INFO [GadgetStreamController.cpp:190] Stream is closed
```

There doesn't seem to be an error, but I am not sure since these logs are a bit hard to read for me. I don't know what would be the best way for me to investigate this bug.

**Comment by zaccharieramzi** [comment]
Actually I just tried with the shepp logan phantom (from `ismrmrd_generate_cartesian_shepp_logan`), and it seems that the dtype uint16 is not necessarily an error.

I get the following image with GRAPPA reconstruction with the phantom:
![grappa_phantom_gadgetron_recon](https://user-images.githubusercontent.com/6387497/87549054-9cc4b380-c6ad-11ea-944a-d272e402e9a7.png)

And the image is of dtype uint16.

Maybe the real question is: how can I get GRAPPA reconstruction without having to use uint16?

**Comment by zaccharieramzi** [comment]
So in the end I tried to scale my data beforehand and I got a result that is not all 0s, but still all uint16.
Another problem is that it's not a satisfactory reconstruction as I would have expected but I guess it's for another issue.

This is the image I get as a result:
![grappa_gadgetron_recon](https://user-images.githubusercontent.com/6387497/87550125-00031580-c6af-11ea-9aa2-aad38fb1b117.png)

As you can see it's almost the same as with the default reconstruction.

**Comment by zaccharieramzi** [comment]
I see there is a `grappa_float_cpu.xml` configuration so maybe there is a solution there, but when running it, I don't have any output files without any error on the client side or on the server side.

Here are the server-side logs:
```
07-15 13:55:39.669 INFO [GadgetStreamController.cpp:48] Connection from 127.0.0.1:36338
07-15 13:55:39.669 INFO [GadgetStreamController.cpp:202] Running configuration: /usr/local/share/gadgetron/config/grappa_float_cpu.xml
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:226] Found 2 readers
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:227] Found 1 writers
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:228] Found 9 gadgets
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:244] --Found reader declaration
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:245]   Reader dll: gadgetron_mricore
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:246]   Reader class: GadgetIsmrmrdAcquisitionMessageReader
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:247]   Reader slot: 1008
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:244] --Found reader declaration
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:245]   Reader dll: gadgetron_mricore
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:246]   Reader class: GadgetIsmrmrdWaveformMessageReader
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:247]   Reader slot: 1026
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:277] --Found writer declaration
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:278]   Writer dll: gadgetron_mricore
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:279]   Writer class: MRIImageWriter
07-15 13:55:39.670 INFO [GadgetStreamController.cpp:280]   Writer slot: 1022
07-15 13:55:39.670 DEBUG [GadgetStreamController.cpp:296] Processing 9 gadgets in reverse order
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:311]   Gadget Name: ImageFinish
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:313]   Gadget class: ImageFinishGadget
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:311]   Gadget Name: Extract
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:313]   Gadget class: ExtractGadget
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:311]   Gadget Name: GrappaUnmixing
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_grappa
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:313]   Gadget class: GrappaUnmixingGadget
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:311]   Gadget Name: Grappa
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_grappa
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:313]   Gadget class: GrappaGadget
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 2
07-15 13:55:39.671 INFO [GadgetStreamController.cpp:335] Setting parameter target_coils = 8
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:335] Setting parameter use_gpu = false
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:311]   Gadget Name: RemoveROOversampling
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:313]   Gadget class: RemoveROOversamplingGadget
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:311]   Gadget Name: AsymmetricEcho
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:313]   Gadget class: AsymmetricEchoAdjustROGadget
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:311]   Gadget Name: CoilReduction
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:313]   Gadget class: CoilReductionGadget
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 1
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:335] Setting parameter coils_out = 16
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:311]   Gadget Name: PCA
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 13:55:39.672 INFO [GadgetStreamController.cpp:313]   Gadget class: PCACoilGadget
07-15 13:55:39.673 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 13:55:39.673 INFO [GadgetStreamController.cpp:310] --Found gadget declaration
07-15 13:55:39.673 INFO [GadgetStreamController.cpp:311]   Gadget Name: NoiseAdjust
07-15 13:55:39.673 INFO [GadgetStreamController.cpp:312]   Gadget dll: gadgetron_mricore
07-15 13:55:39.673 INFO [GadgetStreamController.cpp:313]   Gadget class: NoiseAdjustGadget
07-15 13:55:39.673 INFO [GadgetStreamController.cpp:328]   Gadget parameters: 0
07-15 13:55:39.673 INFO [GadgetStreamController.cpp:356] Gadget Stream configured
07-15 13:55:39.673 DEBUG [NoiseAdjustGadget.cpp:58] Folder to store noise dependencies is /tmp/gadgetron/
07-15 13:55:39.673 DEBUG [NoiseAdjustGadget.cpp:63] NoiseAdjustGadget::perform_noise_adjust_ is 1
07-15 13:55:39.673 DEBUG [NoiseAdjustGadget.cpp:66] NoiseAdjustGadget::pass_nonconformant_data_ is 0
07-15 13:55:39.673 DEBUG [NoiseAdjustGadget.cpp:75] receiver_noise_bandwidth_ is 0.793000
07-15 13:55:39.674 DEBUG [NoiseAdjustGadget.cpp:84] Measurement ID is 67041_36245722_36245731_3604
07-15 13:55:39.674 DEBUG [PCACoilGadget.cpp:65] Number of uncombined channels (present_uncombined_channels) set to 0
07-15 13:55:39.674 DEBUG [CoilReductionGadget.cpp:66] Coil reduction from 15 to 15
07-15 13:55:39.675 DEBUG [AsymmetricEchoAdjustROGadget.cpp:23] max RO for encoding space  0 : 640
07-15 13:55:39.675 DEBUG [RemoveROOversamplingGadget.cpp:43] RemoveROOversamplingGadget:omp_set_num_threads(1) ... 
07-15 13:55:39.675 DEBUG [GrappaGadget.cpp:71] Dimensions 640, 368, 1, 15, 35
07-15 13:55:39.705 DEBUG [Gadget.h:174] Gadget (NoiseAdjust) Close Called with flags = 1
07-15 13:55:39.706 DEBUG [Gadget.h:184] Gadget (NoiseAdjust) waiting for thread to finish
07-15 13:55:39.706 DEBUG [Gadget.h:174] Gadget (NoiseAdjust) Close Called with flags = 0
07-15 13:55:39.706 DEBUG [Gadget.h:186] Gadget (NoiseAdjust) thread finished
07-15 13:55:39.706 DEBUG [Gadget.h:123] Shutting down Gadget (NoiseAdjust)
07-15 13:55:39.706 DEBUG [Gadget.h:174] Gadget (PCA) Close Called with flags = 1
07-15 13:55:39.706 DEBUG [Gadget.h:184] Gadget (PCA) waiting for thread to finish
07-15 13:55:39.706 DEBUG [Gadget.h:174] Gadget (PCA) Close Called with flags = 0
07-15 13:55:39.706 DEBUG [Gadget.h:186] Gadget (PCA) thread finished
07-15 13:55:39.706 DEBUG [Gadget.h:123] Shutting down Gadget (PCA)
07-15 13:55:39.706 DEBUG [Gadget.h:174] Gadget (CoilReduction) Close Called with flags = 1
07-15 13:55:39.706 DEBUG [Gadget.h:184] Gadget (CoilReduction) waiting for thread to finish
07-15 13:55:39.706 DEBUG [Gadget.h:174] Gadget (CoilReduction) Close Called with flags = 0
07-15 13:55:39.706 DEBUG [Gadget.h:186] Gadget (CoilReduction) thread finished
07-15 13:55:39.706 DEBUG [Gadget.h:123] Shutting down Gadget (CoilReduction)
07-15 13:55:39.706 DEBUG [Gadget.h:174] Gadget (AsymmetricEcho) Close Called with flags = 1
07-15 13:55:39.706 DEBUG [Gadget.h:184] Gadget (AsymmetricEcho) waiting for thread to finish
07-15 13:55:39.706 DEBUG [Gadget.h:174] Gadget (AsymmetricEcho) Close Called with flags = 0
07-15 13:55:39.706 DEBUG [Gadget.h:186] Gadget (AsymmetricEcho) thread finished
07-15 13:55:39.707 DEBUG [Gadget.h:123] Shutting down Gadget (AsymmetricEcho)
07-15 13:55:39.707 DEBUG [Gadget.h:174] Gadget (RemoveROOversampling) Close Called with flags = 1
07-15 13:55:39.707 DEBUG [Gadget.h:184] Gadget (RemoveROOversampling) waiting for thread to finish
07-15 13:55:39.707 DEBUG [Gadget.h:174] Gadget (RemoveROOversampling) Close Called with flags = 0
07-15 13:55:39.707 DEBUG [Gadget.h:186] Gadget (RemoveROOversampling) thread finished
07-15 13:55:39.707 DEBUG [Gadget.h:123] Shutting down Gadget (RemoveROOversampling)
07-15 13:55:39.707 DEBUG [Gadget.h:174] Gadget (Grappa) Close Called with flags = 1
07-15 13:55:39.707 DEBUG [Gadget.h:184] Gadget (Grappa) waiting for thread to finish
07-15 13:55:39.707 DEBUG [Gadget.h:174] Gadget (Grappa) Close Called with flags = 0
07-15 13:55:39.707 DEBUG [GrappaGadget.cpp:37] Shutting down GRAPPA Gadget
07-15 13:55:39.707 DEBUG [Gadget.h:186] Gadget (Grappa) thread finished
07-15 13:55:39.707 DEBUG [GrappaGadget.cpp:37] Shutting down GRAPPA Gadget
07-15 13:55:39.707 DEBUG [Gadget.h:123] Shutting down Gadget (Grappa)
07-15 13:55:39.707 DEBUG [Gadget.h:174] Gadget (GrappaUnmixing) Close Called with flags = 1
07-15 13:55:39.707 DEBUG [Gadget.h:184] Gadget (GrappaUnmixing) waiting for thread to finish
07-15 13:55:39.707 DEBUG [Gadget.h:174] Gadget (GrappaUnmixing) Close Called with flags = 0
07-15 13:55:39.707 DEBUG [Gadget.h:186] Gadget (GrappaUnmixing) thread finished
07-15 13:55:39.707 DEBUG [Gadget.h:123] Shutting down Gadget (GrappaUnmixing)
07-15 13:55:39.707 DEBUG [Gadget.h:174] Gadget (Extract) Close Called with flags = 1
07-15 13:55:39.707 DEBUG [Gadget.h:184] Gadget (Extract) waiting for thread to finish
07-15 13:55:39.708 DEBUG [Gadget.h:174] Gadget (Extract) Close Called with flags = 0
07-15 13:55:39.708 DEBUG [Gadget.h:186] Gadget (Extract) thread finished
07-15 13:55:39.708 DEBUG [Gadget.h:123] Shutting down Gadget (Extract)
07-15 13:55:39.708 DEBUG [Gadget.h:174] Gadget (ImageFinish) Close Called with flags = 1
07-15 13:55:39.708 DEBUG [Gadget.h:184] Gadget (ImageFinish) waiting for thread to finish
07-15 13:55:39.708 DEBUG [Gadget.h:174] Gadget (ImageFinish) Close Called with flags = 0
07-15 13:55:39.708 DEBUG [Gadget.h:186] Gadget (ImageFinish) thread finished
07-15 13:55:39.708 DEBUG [Gadget.h:123] Shutting down Gadget (ImageFinish)
07-15 13:55:39.708 DEBUG [EndGadget.cpp:12] Close called in EndGadget with flags 1
07-15 13:55:39.708 DEBUG [EndGadget.cpp:23] Calling close in base class  with flags 1
07-15 13:55:39.708 DEBUG [Gadget.h:174] Gadget (EndGadget) Close Called with flags = 1
07-15 13:55:39.708 DEBUG [Gadget.h:184] Gadget (EndGadget) waiting for thread to finish
07-15 13:55:39.708 DEBUG [EndGadget.cpp:12] Close called in EndGadget with flags 0
07-15 13:55:39.708 DEBUG [EndGadget.cpp:23] Calling close in base class  with flags 0
07-15 13:55:39.708 DEBUG [Gadget.h:174] Gadget (EndGadget) Close Called with flags = 0
07-15 13:55:39.708 DEBUG [Gadget.h:186] Gadget (EndGadget) thread finished
07-15 13:55:39.708 DEBUG [Gadget.h:123] Shutting down Gadget (EndGadget)
07-15 13:55:39.708 DEBUG [GadgetStreamController.cpp:80] Stream closed
07-15 13:55:39.709 DEBUG [GadgetStreamController.cpp:81] Closing writer task
07-15 13:55:39.709 DEBUG [GadgetStreamController.cpp:83] Writer task closed
07-15 13:55:39.709 ERROR [GadgetStreamController.cpp:74] GadgetStreamController, unable to read message identifier
07-15 13:55:39.709 DEBUG [GadgetStreamController.cpp:159] handle_close called
07-15 13:55:39.709 INFO [GadgetStreamController.cpp:164] Shutting down stream and closing up shop...
07-15 13:55:39.709 INFO [GadgetStreamController.cpp:190] Stream is closed
```

**Comment by valeryozenne** [comment]
Hi,

Most of the reconstruction are configured to send back uint16 data as this is required by the scanner (in order to match the DICOM format). To get data in float , this can be done by creating a new configuration file (a copy of Generic Cartesian Grappa for instance ) and commenting the gadget "FloattoShort".  You can directly call this configuration file with the option -C .using the gadgetron_ismsmrd_client.

Valéry

**Comment by zaccharieramzi** [comment]
Makes total sense, thanks for this explanation.

**Comment by zaccharieramzi** [comment]
Interestingly someone seems to have the same problem [here](https://gadgetron.discourse.group/t/grappa-reconstruction-runs-without-errors-but-results-with-an-empty-image/131). I will answer once my gadgetron discourse account is approved.

**Comment by dchansen** [comment]
I'm not seeing any pending discourse applications. Did you get approved?

**Comment by zaccharieramzi** [comment]
Yes ! Thanks for checking.
